### DISCUSSION TITLE
Where can I find a good example of how to close down with async pymodbus.server.ServerAsyncStop()?

### DISCUSSION THREAD

**Discussion by nildude** [pymodbus-dev/pymodbus#1482] [Help] [answered]
Hi community, 

Where can I find a good example of how to close down with async pymodbus.server.ServerAsyncStop()? I want to run the server for a fixed amount of time and not running forever? How is the stop function working? I cannot seem to find a good example. 

Best regards, 
Lars Henrik

**Comment by janiversen** [comment, marked as answer]
We have tests where it is done a lot, have a look at test/test_examples.py.

But really in async it is very simple just do "await ServerAsyncStop()"
